integrations/tensorflow/e2e/keras/layers/layers_test.py

```diff
@@ -0,0 +1,270 @@
+    'DepthwiseConv2D': dict(kernel_size=3),
+    'Dot': dict(axes=(1, 2)),
+    'Dropout': dict(rate=0.9),
+    'ELU': dict(),
+    'Embedding': dict(input_dim=4, output_dim=2),
+    'Flatten': dict(),
+    'GRU': dict(units=4, return_sequences=True),
+    'GRUCell': dict(units=4),
+    'GaussianDropout': dict(rate=0.9),
+    'GaussianNoise': dict(stddev=1.0),
+    'GlobalAveragePooling1D': dict(),
+    'GlobalAveragePooling2D': dict(),
+    'GlobalAveragePooling3D': dict(),
+    'GlobalMaxPool1D': dict(),
+    'GlobalMaxPool2D': dict(),
+    'GlobalMaxPool3D': dict(),
+    'InputLayer': dict(),
+    'LSTM': dict(units=4, return_sequences=True),
+    'LSTMCell': dict(units=4),
+    'Lambda': dict(function=lambda x: x**2),
+    'Layer': dict(),
+    'LayerNormalization': dict(),
+    'LeakyReLU': dict(),
+    'LocallyConnected1D': dict(filters=4, kernel_size=3),
+    'LocallyConnected2D': dict(filters=4, kernel_size=3),
+    'Masking': dict(),
+    'MaxPool1D': dict(),
+    'MaxPool2D': dict(),
+    'MaxPool3D': dict(),
+    'Maximum': dict(),
+    'Minimum': dict(),
+    'MultiHeadAttention': dict(num_heads=2, key_dim=3),
+    'Multiply': dict(),
+    'PReLU': dict(),
+    'Permute': dict(dims=(3, 1, 2)),
+    'ReLU': dict(),
+    'RepeatVector': dict(n=3),
+    'Reshape': dict(target_shape=[1, 1, 1] + RANK_3_INPUT[1:]),
+    'SeparableConv1D': dict(filters=4, kernel_size=3),
+    'SeparableConv2D': dict(filters=4, kernel_size=3),
+    'SimpleRNN': dict(units=4, return_sequences=True),
+    'SimpleRNNCell': dict(units=4),
+    'Softmax': dict(),
+    'SpatialDropout1D': dict(rate=0.9),
+    'SpatialDropout2D': dict(rate=0.9),
+    'SpatialDropout3D': dict(rate=0.9),
+    'Subtract': dict(),
+    'ThresholdedReLU': dict(),
+    'UpSampling1D': dict(),
+    'UpSampling2D': dict(),
+    'UpSampling3D': dict(),
+    'ZeroPadding1D': dict(),
+    'ZeroPadding2D': dict(),
+    'ZeroPadding3D': dict(),
+}
+
+flags.DEFINE_string('layer', 'Dense',
+                    f'One of {list(LAYER_TO_INPUT_SHAPES.keys())}.')
+flags.DEFINE_bool(
+    'dynamic_batch', False,
+    'Whether or not to compile the layer with a dynamic batch size.')
+
+
+def get_input(shape: Sequence[int]) -> tf.keras.layers.Input:
+  batch_size = None if FLAGS.dynamic_batch else shape[0]
+  return tf.keras.layers.Input(batch_size=batch_size, shape=shape[1:])
+
+
+def normalize(inputs: Sequence[Any]) -> Union[Any, Sequence[Any]]:
+  """Unpacks inputs if it has length one."""
+  return inputs[0] if len(inputs) == 1 else inputs
+
+
+class KerasLayersModule(tf.Module):
+
+  def __init__(self):
+    super().__init__()
+    layer_class = getattr(tf.keras.layers, FLAGS.layer)
+    kwargs = LAYER_TO_KWARGS[FLAGS.layer]
+    input_shapes = LAYER_TO_INPUT_SHAPES[FLAGS.layer]
+
+    # Create a wrapped keras layer.
+    inputs = normalize([get_input(shape) for shape in input_shapes])
+    if FLAGS.layer == 'MultiHeadAttention':
+      outputs = layer_class(**kwargs)(*inputs)
```

Comment:
:-/ Special-casing this here seems not great

Reply:
Yeah, this is a bug in their API :/

Reply:
Can you add a link/TODO then? :-)